flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -148,9 +147,6 @@ else if (pKeys instanceof Keys.SelectorFunctionKeys) {
 			}
 			
 		} 
-		else if (pMethod == PartitionMethod.RANGE) {
```

Comment:
I see your point. It's probably OK to leave out this check. `sortPartition()` is doing the same check and would fail if the same non-sortable key is used.